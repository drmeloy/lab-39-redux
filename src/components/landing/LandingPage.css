.Landing {
  display: flex;
  flex-direction: column;
  align-items: center;

  h1 {
    margin: 30px;
    font-size: 4rem;
  }

  p {
    margin-bottom: 30px;
    font-size: 1.2rem;
    text-align: center;
  }

  button {
    height: 55px;
    width: 165px;
    border-radius: 5px;
    font-size: 16px;
    background: yellowgreen;
  }

  button:hover {
    background: #57741b;
    color: white;
  }
}