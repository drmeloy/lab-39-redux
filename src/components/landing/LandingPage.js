import React from 'react';
import { Link } from 'react-router-dom';
import styles from './LandingPage.css';
import { useDispatch } from 'react-redux';
import { clearAll } from '../../actions/moodActions';

const LandingPage = () => {
  const dispatch = useDispatch();

  return (
    <section className={styles.Landing}>
      <h1>Welcome to Facey Face!</h1>
      <p>You'll have 5 seconds to make your face the faciest face ever. Prepare yourself. Deep breath. When you're ready, click below to start!</p>
      <Link to='/faceyface'>
        <button onClick={dispatch(clearAll())}>Click to begin!</button>
      </Link>
    </section>
  );
}

export default LandingPage;
