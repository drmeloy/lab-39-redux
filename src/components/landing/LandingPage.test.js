import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from './LandingPage';

describe('LandingPage component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<LandingPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
