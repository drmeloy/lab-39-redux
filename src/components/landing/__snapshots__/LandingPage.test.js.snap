// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component matches snapshot 1`] = `
<section
  className="Landing"
>
  <h1>
    Welcome to Facey Face!
  </h1>
  <p>
    You'll have 5 seconds to make your face the faciest face ever. Prepare yourself. Deep breath. When you're ready, click below to start!
  </p>
  <Link
    to="/faceyface"
  >
    <button>
      Click to begin!
    </button>
  </Link>
</section>
`;
